### ISSUE TITLE
Encoding error when generating schedule

### ISSUE THREAD

**Issue by irodimus** [justinemter/pseudo-channel#26]
This happens from time to time due to the naming of the video files.

```
#### Adding media to db Traceback (most recent call last):
  File "PseudoChannel.py", line 1033, in <module>
    pseudo_channel.generate_daily_schedule()
  File "PseudoChannel.py", line 822, in generate_daily_schedule
    self.db.add_media_to_daily_schedule(entry)
  File "/Applications/pseudo-channel/src/PseudoChannelDatabase.py", line 258, in add_media_to_daily_schedule
    print "#### Adding media to db", media.title, media.start_time
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 11: ordinal not in range(128)
```

character u'\u2019 can be different. Doesn't happen often.

**Comment by justinemter** [comment]
So, unfortunately, the downside of using xml is you have to manually encode special characters like '&' and even apostrophe's etc. You can try this service here: http://coderstoolbox.net/string/#!encoding=xml&action=encode&charset=us_ascii

...I've never used it but it's supposed to just run through your xml and convert any illegal characters to ascii. I look forward to getting this app in a good place first, then I will create a much easier method.

**Comment by justinemter** [comment]
I also added another fix to 'develop', it may work now. :)

**Comment by justinemter** [comment]
Hey there, out of curiosity - did you get it to work?

**Comment by irodimus** [comment]
I haven't run into any issues and there are names in the schedule that have ampersands, commas, and apostrophes, but the several times I did when you implemented the fix did not give an error so it looks like it may have resolved it.

**Comment by irodimus** [comment]
I did run into one issue when generating the schedule, but it is probably not related. 

```
Traceback (most recent call last):
  File "PseudoChannel.py", line 1033, in <module>
    pseudo_channel.generate_daily_schedule()
  File "PseudoChannel.py", line 802, in generate_daily_schedule
    previous_episode.overlap_max
  File "PseudoChannel.py", line 572, in calculate_start_time
    return newStartTime.strftime('%I:%M:%S %p')
AttributeError: 'str' object has no attribute 'strftime'
```

**Comment by justinemter** [comment]
Hmm, so the "overlap-max" is a setting that is in the xml but I never integrated it within the app. For now I'd leave that "overlap-max" attribute exactly how I have it in the xml. I'm thinking you took it out of some time entries and somewhere (line 572) I am expecting it... So yeah, for now leave that in the <time> entries as-is. The app will automatically run that "-g" flag at midnight and will error / stop running. Just restore it in the xml and rerun this: "python PseudoChannel.py -xml". Curious to hear how it goes. :)

**Comment by irodimus** [comment]
I actually just copy and pasted the line

`<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">6:00 AM</time>
`

from your schedule.xml in the repository to make my schedule.xml look like this

```
<?xml version="1.0" encoding="UTF-8"?>
<schedule>
	<everyday> 
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">6:00 AM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">7:30 AM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">9:00 AM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">10:30 AM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">12:00 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">1:30 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">3:00 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">4:30 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">6:00 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">7:30 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">9:00 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">10:30 PM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">12:00 AM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">1:30 AM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">3:00 AM</time>
		<time title="random" type="movie" strict-time="false" time-shift="5" overlap-max="">4:30 AM</time>
	</everyday>
	<mondays></mondays>
	<tuesdays></tuesdays>
	<wednesdays></wednesdays>
	<thursdays></thursdays>
	<fridays></fridays>
	<saturdays></saturdays>
	<sundays></sundays>
	<weekends></weekends>
	<weekdays></weekdays>
</schedule>
```
you weren't showing anything in the overlap-max value. Is there something supposed to be there?

**Comment by irodimus** [comment]
Man, I spoke too soon. Just got this when messing when generating schedule

```
Traceback (most recent call last):
  File "PseudoChannel.py", line 1041, in <module>
    pseudo_channel.show_schedule()
  File "PseudoChannel.py", line 906, in show_schedule
    print "+++++", str(i + 1)+".", entry[8], entry[11], entry[6], " - ", entry[3]
UnicodeEncodeError: 'ascii' codec can't encode character u'\xb2' in position 5: ordinal not in range(128)
```

**Comment by justinemter** [comment]
> you weren't showing anything in the overlap-max value. Is there something supposed to be there?
It could also be my time as I've spaced everything 1:30 so the time different is always off when it adjusts. Still getting the hang of the XML attributes.

No that looks good! I jumped to conclusions. I'll have to look into that. 

So I'm going to fix the "-s" real quick (or try to) but do know that the ascii issues so far are only breaking in the terminal output... I believe the app itself would still run.
